pyocd/flash/loader.py

```diff
@@ -197,17 +198,24 @@ def _program_hex(self, file_obj, **kwargs):
                 LOG.warning("Failed to add data chunk: %s", e)
 
     def _program_elf(self, file_obj, **kwargs):
-        """! ELF format loader"""
-        elf = ELFBinaryFile(file_obj, self._session.target.memory_map)
-        for section in elf.sections:
-            if ((section.type == 'SHT_PROGBITS')
-                    and ((section.flags & (SH_FLAGS.SHF_ALLOC | SH_FLAGS.SHF_WRITE)) == SH_FLAGS.SHF_ALLOC)
-                    and (section.length > 0)
-                    and (section.region.is_flash)):
-                LOG.debug("Writing section %s", repr(section))
-                self._loader.add_data(section.start, section.data)
+        elf = ELFFile(file_obj)
+        for segment in elf.iter_segments():
+            if segment.header.p_type == 'PT_LOAD' and segment.header.p_filesz != 0:
+                addr = segment['p_paddr']
+                if sys.version_info[0] == 3:
```

Comment:
I think you can use `bytearray(segment.data())` here instead of switching on the Python version.

If not, please use the helpers in `pyocd.utility.conversion` or `pyocd.utility.compatibility` for data conversion. (Or add one if necessary.)

Reply:
Done